x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/find_rules_route.ts

```diff
@@ -62,12 +65,15 @@ export const findRulesRoute = (
         });
         const alertIds = rules.data.map((rule) => rule.id);
 
-        const ruleStatuses = await execLogClient.findBulk({
-          ruleIds: alertIds,
-          logsCount: 1,
-          spaceId: context.securitySolution.getSpaceId(),
-        });
-        const transformed = transformFindAlerts(rules, ruleStatuses);
+        const [ruleStatuses, ruleActions] = await Promise.all([
+          execLogClient.findBulk({
+            ruleIds: alertIds,
+            logsCount: 1,
+            spaceId: context.securitySolution.getSpaceId(),
+          }),
+          legacyGetBulkRuleActionsSavedObject({ alertIds, savedObjectsClient }),
+        ]);
```

Comment:
we should add a `.catch` to this promise.all *or* add a try-catch block inside of [`legacyGetBulkRuleActionsSavedObject`](https://github.com/elastic/kibana/pull/112869/files#diff-716c4f1364562dcc82814113cccb3aa0448d3ee14241b2a1925aee6a368b1346R30) to cover the case where something like the `find` api on the saved objects client throws an error.

Reply:
Below this chunk for the route we have a try/catch where the whole thing will fail if one or the other of the actions cannot be retrieved. I don't think the `find` API will cause an error within its self.

This was the original behavior before fwiw in 7.15.0 and 7.14.0.